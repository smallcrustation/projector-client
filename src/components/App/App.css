#root {
  max-width: 800px;
  margin: 0 auto;
}

body {
}

form {
  display: flex;
  flex-direction: column;
}

input {
  width: 80%;
  border: 1px lightgray solid;
  padding: 10px;
}

label {
  display: block;
}

ul {
  list-style: none;
}

button {
  border: 3px solid black;
  box-shadow: -3px 5px;
  margin: 20px;
  background-color: white;
  padding: 10px 15px;
  font-size: 20px;
  outline-color: slategrey;
  transition: 0.3s;
}

a:active {
  background-color: lightgray;
}

input[type='submit'] {
  border: 3px solid black;
  box-shadow: -3px 5px;
  margin: 20px;
  background-color: white;
  padding: 15px 10px;
  font-size: 20px;
  transition: 0.3s;
}

input:focus,
select:focus,
textarea:focus,
button:focus a:focus {
  outline-color: slategrey;
}

form {
  margin: 3em auto 5px auto;
  display: flex;
  flex-direction: column;
  max-width: 500px;
  text-align: center;
  /* align-items: center;
  justify-content: center; */
}

label {
  font-size: 30px;
  text-align: left;
}

input {
  font-size: 27px;
  margin: 15px;
  border: 1px solid black;
  text-align: center;
}

a {
  color: black;
  outline-color: slategrey;
  text-decoration: none;
  transition: 0.3s;
}

h3 {
  /* text-align: center; */
  font-size: 29px;
  margin: 10px 0 15px 0;
  padding: 0 20px 3px;
  background-image: linear-gradient(to right, black, black);
  /* background-position: bottom center ; */
  background-position: bottom left;
  background-repeat: no-repeat;
  background-size: 8em 2px;
}

a:visited {
  color: black;
}

.error {
  color: red;
}

.thick-brdr {
  border-width: 5px;
}

.dollar-input {
  font-size: 27px;
  text-align: center;
  margin: 10px;
}

.low-header {
  /* padding-top: 40px; */
  margin-top: 60px;
}

.link-btn {
  border: 3px solid black;
  box-shadow: -3px 5px;
  margin: 20px;
  background-color: white;
  padding: 10px 15px;
  font-size: 20px;
  outline-color: slategrey;
  transition: 0.3s;
}

.loading-dots:after {
  content: ' .';
  animation: dots 1s steps(5, end) infinite;
}

@keyframes dots {
  0%,
  20% {
    color: rgba(0, 0, 0, 0);
    text-shadow: 0.25em 0 0 rgba(0, 0, 0, 0), 0.5em 0 0 rgba(0, 0, 0, 0);
  }
  40% {
    color: black;
    text-shadow: 0.25em 0 0 rgba(0, 0, 0, 0), 0.5em 0 0 rgba(0, 0, 0, 0);
  }
  60% {
    text-shadow: 0.25em 0 0 black, 0.5em 0 0 rgba(0, 0, 0, 0);
  }
  80%,
  100% {
    text-shadow: .25em 0 0 black, .5em 0 0 black;
  }
}

/* 
.ripple {
  background-position: center;
  transition: background 0.5s;
}
.ripple:hover {
  background: lightgray radial-gradient(circle, transparent 1%, white 1%) center/15000%;
}
.ripple:active {
  background-color: lightgray;
  background-size: 100%;
  transition: background .2s;
} */

/* .cbutton--effect-stana .cbutton__helper {
	position: absolute;
	top: 50%;
	left: 50%;
	margin: -40px 0 0 -40px;
	width: 80px;
	height: 80px;
	opacity: 0;
	pointer-events: none;
}

.cbutton--effect-stana line {
	fill: none;
	stroke-width: 2;
}

.cbutton--box-color-7 line {
	stroke: #0077ff;
}

.cbutton--box-color-8 line {
	stroke: #7bff00;
}

.cbutton--box-color-9 line {
	stroke: #fff;
}

.clip-ring {
    -webkit-transform-origin: 50% 50%;
    transform-origin: 50% 50%;
}

.cbutton--effect-stana.cbutton--click .cbutton__helper {
	-webkit-animation: anim-effect-stana 0.4s ease-in forwards;
	animation: anim-effect-stana 0.4s ease-in forwards;
}

.cbutton {
	position: relative;
	display: inline-block;
	margin: 1em;
	padding: 0;
	border: none;
	background: none;
	color: #286aab;
	font-size: 1.4em;
	overflow: visible;
	-webkit-transition: color 0.7s;
	transition: color 0.7s;
	-webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.cbutton.cbutton--click,
.cbutton:focus {
	outline: none;
	color: #3c8ddc;
}

.cbutton__icon {
	display: block;
}

.cbutton__text {
	position: absolute;
	opacity: 0;
	pointer-events: none;
}

.cbutton::after {
	position: absolute;
	top: 50%;
	left: 50%;
	margin: -35px 0 0 -35px;
	width: 70px;
	height: 70px;
	border-radius: 50%;
	content: '';
	opacity: 0;
	pointer-events: none;
}

.cbutton--box {
	width: 24px;
	height: 24px;
	border: 4px solid rgb(22, 35, 47);
}

.cbutton--box-color-1 {
	background: #ff8b00;
}

.cbutton--box-color-2 {
	background: #b61854;
}

.cbutton--box-color-3 {
	background: #1d5cab;
}

.cbutton--box-color-4 {
	background: #f2438c;
}

.cbutton--box-color-5 {
	background: #ff6414;
}

.cbutton--box-color-6 {
	background: #5ed500;
}

.cbutton--box-color-7 {
	background: #0077ff;
}

.cbutton--box-color-8 {
	background: #7bff00;
}

.cbutton--box-color-9 {
	background: #fff;
} 

@-webkit-keyframes anim-effect-stana {
	0%, 100% {
		opacity: 1;
	}
	50% {
		opacity: 0.3;

		-webkit-animation-timing: ease-out;
		animation-timing: ease-out;
	}
}

@keyframes anim-effect-stana {
	0%, 100% {
		opacity: 1;
	}
	50% {
		opacity: 0.3;

		-webkit-animation-timing: ease-out;
		animation-timing: ease-out;
	}
}

.cbutton--effect-stana.cbutton--click .clip-ring {
	-webkit-transition: -webkit-transform 0.4s ease-in-out;
    transition: transform 0.4s ease-in-out;
	-webkit-transform: scale3d(3, 3, 1);
	transform: scale3d(3, 3, 1);
} */

@media only screen and (min-width: 800px) {
  /* desktop */
  button:hover,
  input[type='submit']:hover {
    cursor: pointer;
  }

  button:hover {
    background-color: lightgray;
  }

  input[type='submit']:hover {
    background-color: lightgreen;
  }

  a:hover {
    background-color: lightgray;
  }
}
