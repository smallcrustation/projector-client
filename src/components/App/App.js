import React from 'react'
import { Switch, Route } from 'react-router-dom'

import Header from '../Header/Header'
import Nav from '../Nav/Nav'
import Landing from '../../routes/LandingPage/LandingPage'
import CreateAccount from '../../routes/CreateAccountPage/CreateAccountPage'
import Login from '../../routes/LoginPage/LoginPage'
import NotFound from '../../routes/NotFoundPage/NotFoundPage'

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Nav />
        <main className="App__main">
          <Switch>
            <Route exact path={'/'} component={Landing} />
            <Route exact path={'/createAccount'} component={CreateAccount} />
            <Route exact path={'/login'} component={Login} />
            <Route component={NotFound} />
          </Switch>
        </main>
      </div>
    )
  }
}

export default App
