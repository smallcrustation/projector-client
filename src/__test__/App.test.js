import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'

import App from '../components/App/App'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      div
    )

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
