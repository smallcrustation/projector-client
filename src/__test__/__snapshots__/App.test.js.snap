// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="Header"
  >
    <a
      className="Header__h1"
      href="/"
      onClick={[Function]}
    >
      <h1
        className="Header__h1"
      >
        Projector
      </h1>
    </a>
    <nav
      className="logged-out-nav nav-bar"
    >
      <a
        className="link-btn"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        className="link-btn"
        href="/createAccount"
        onClick={[Function]}
      >
        Create
      </a>
    </nav>
  </header>
  <main
    className="App__main"
  >
    <section
      className="LandingPage"
    >
      <h2
        className="LandingPage__h3"
      >
        Welcome to Projector
      </h2>
      <article>
        <p
          className="LandingPage__p"
        >
          Projector is an app designed for you to track your project by it's funding.
          <br />
          If this is your first time using the app please click 'Create' to create an account.
          <br />
          (demo login user:user pass:password)
        </p>
        <a
          className="link-btn"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          className="link-btn"
          href="/createAccount"
          onClick={[Function]}
        >
          Create
        </a>
      </article>
    </section>
  </main>
</div>
`;
